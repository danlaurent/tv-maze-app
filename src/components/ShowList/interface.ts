import { TShowList } from '../../interface/shows';

export interface IShowList {
  shows: TShowList | null;
  onMoviePress?: (showId: number) => void;
  onEndReached?: () => void;
  testID?: string;
}
