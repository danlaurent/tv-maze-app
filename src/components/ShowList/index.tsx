import React from 'react';
import { FlatList, View, StyleSheet } from 'react-native';
import ShowListItem from '../ShowListItem';
import { IShowList } from './interface';

const styles = StyleSheet.create({
  itemContainer: {
    paddingVertical: 8,
  },
});

const ShowList = ({ shows, onEndReached, onMoviePress, testID }: IShowList) => {
  if (shows && shows.length > 0) {
    return (
      <FlatList
        testID={testID}
        data={shows}
        renderItem={({ item }) => {
          return (
            <View style={styles.itemContainer}>
              <ShowListItem
                show={item}
                onPress={onMoviePress}
                testID={`${testID}_${item.id}`}
              />
            </View>
          );
        }}
        onEndReachedThreshold={0.9}
        onEndReached={onEndReached}
        keyExtractor={(item) => item.id.toString()}
        contentContainerStyle={{ padding: 20 }}
      />
    );
  }
  return null;
};

export default ShowList;
