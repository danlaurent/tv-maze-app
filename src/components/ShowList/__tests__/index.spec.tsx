import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';
import ShowList from '..';

const showsMock: any = [
  {
    id: 1,
    name: 'Test Movie',
    rating: {
      average: 8.0,
    },
    image: {
      medium: 'https://via.placeholder.com/150',
    },
    genres: ['Test'],
  },
];

const emptyShowsMock: any = [];

describe('ShowList', () => {
  let tree: RenderAPI;
  beforeEach(() => {
    tree = render(<ShowList shows={showsMock} />);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When there is no shows', () => {
    beforeEach(() => {
      tree = render(<ShowList testID='testShowList' shows={emptyShowsMock} />);
    });

    it('do not render the show list', () => {
      expect(tree.queryByTestId('testShowList')).toBeNull();
    });
  });
});
