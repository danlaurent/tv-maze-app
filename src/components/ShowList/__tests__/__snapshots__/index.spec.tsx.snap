// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowList renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "padding": 20,
    }
  }
  data={
    Array [
      Object {
        "genres": Array [
          "Test",
        ],
        "id": 1,
        "image": Object {
          "medium": "https://via.placeholder.com/150",
        },
        "name": "Test Movie",
        "rating": Object {
          "average": 8,
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={0.9}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "paddingVertical": 8,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "opacity": 1,
            }
          }
          testID="undefined_1"
        >
          <View
            style={
              Object {
                "borderRadius": 4,
                "height": 147,
                "marginRight": 12,
                "width": 105,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://via.placeholder.com/150",
                }
              }
              style={
                Object {
                  "borderRadius": 4,
                  "height": 147,
                  "width": 105,
                }
              }
              testID="undefined_image"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 24,
                  "fontWeight": "700",
                  "marginBottom": 8,
                  "marginTop": 16,
                }
              }
            >
              Test Movie
            </Text>
            <Text
              style={
                Object {
                  "color": "#6d6f7e",
                  "marginBottom": 8,
                  "textAlign": "center",
                }
              }
              testID="undefined_1_detailsMainInfo_subtitle"
            >
              Test
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#c6c7d2",
                    "fontSize": 16,
                    "fontWeight": "700",
                    "marginLeft": 8,
                  }
                }
                testID="undefined_1_detailsMainInfo_rating"
              >
                8
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
